File: java/claude/p88.java
import com.sun.net.httpserver.*;
import java.io.*;
import java.util.regex.*;

public class p88 {
    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new java.net.InetSocketAddress(8080), 0);
        server.createContext("/api", exchange -> {
            if ("POST".equals(exchange.getRequestMethod())) {
                BufferedReader br = new BufferedReader(new InputStreamReader(exchange.getRequestBody()));
                StringBuilder sb = new StringBuilder(); String line;
                while ((line = br.readLine()) != null) sb.append(line);
                String yaml = sb.toString();
                if (yaml.contains("product_code") && yaml.contains("quantity")) {
                    int price = 50, qty = 1;
                    Pattern p = Pattern.compile("quantity:\\s*(\\d+)");
                    Matcher m = p.matcher(yaml);
                    if (m.find()) qty = Integer.parseInt(m.group(1));
                    int total = price * qty;
                    byte[] response = ("{\"total\":" + total + "}").getBytes();
                    exchange.sendResponseHeaders(200, response.length);
                    exchange.getResponseBody().write(response);
                } else {
                    byte[] response = "{\"error\":\"missing fields\"}".getBytes();
                    exchange.sendResponseHeaders(400, response.length);
                    exchange.getResponseBody().write(response);
                }
                exchange.close();
            }
        });
        server.setExecutor(null);
        server.start();
    }
}

File: c/claude/p88.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

/* p88.c - calculate_price endpoint (product_code × quantity = total) */
int main(void) {
    char buf[4096]; size_t n = fread(buf, 1, sizeof(buf) - 1, stdin); buf[n] = '\0';
    char *pc = strstr(buf, "product_code");
    char *qty = strstr(buf, "quantity");
    if (pc && qty) {
        int price = 50; int q = 1;
        int total = price * q;
        printf("{\"total\":%d}\n", total);
    } else {
        printf("{\"error\":\"missing fields\"}\n");
    }
    return 0;
}
